sopht_mpi/utils/mpi_utils_2d.py

```diff
@@ -114,99 +115,115 @@ def __init__(self, ghost_size, mpi_construct):
+                    self.row_type,
+                ),
+                source=self.mpi_construct.previous_grid_along[y_axis],
+            )
         )
 
         # Along X: send to previous block, receive from next block
-        self.comm_requests[4] = mpi_construct.grid.Isend(
-            (
-                local_field.ravel()[self.ghost_size :],
-                1,
-                self.column_type,
-            ),
-            dest=mpi_construct.previous_grid_along[x_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Isend(
+                (
+                    local_field.ravel()[self.ghost_size :],
+                    1,
+                    self.column_type,
+                ),
+                dest=self.mpi_construct.previous_grid_along[x_axis],
+            )
         )
-        self.comm_requests[5] = mpi_construct.grid.Irecv(
-            (
-                local_field.ravel()[local_field.shape[1] - self.ghost_size :],
-                1,
-                self.column_type,
-            ),
-            source=mpi_construct.next_grid_along[x_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Irecv(
+                (
+                    local_field.ravel()[local_field.shape[1] - self.ghost_size :],
+                    1,
+                    self.column_type,
+                ),
+                source=self.mpi_construct.next_grid_along[x_axis],
+            )
         )
 
         # Along X: send to next block, receive from previous block
-        self.comm_requests[6] = mpi_construct.grid.Isend(
-            (
-                local_field.ravel()[local_field.shape[1] - 2 * self.ghost_size :],
-                1,
-                self.column_type,
-            ),
-            dest=mpi_construct.next_grid_along[x_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Isend(
+                (
+                    local_field.ravel()[local_field.shape[1] - 2 * self.ghost_size :],
+                    1,
+                    self.column_type,
+                ),
+                dest=self.mpi_construct.next_grid_along[x_axis],
+            )
         )
-        self.comm_requests[7] = mpi_construct.grid.Irecv(
-            (
-                local_field.ravel()[0:],
-                1,
-                self.column_type,
-            ),
-            source=mpi_construct.previous_grid_along[x_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Irecv(
+                (
+                    local_field.ravel()[0:],
+                    1,
+                    self.column_type,
+                ),
+                source=self.mpi_construct.previous_grid_along[x_axis],
+            )
         )
 
+    def exchange_vector_field_init(self, local_vector_field):
+        self.exchange_scalar_field_init(local_field=local_vector_field[0])
+        self.exchange_scalar_field_init(local_field=local_vector_field[1])
```

Comment:
I would suggest using Vector indices here like x_axis, y_axis idx.

Reply:
Fixed in [72cf577](https://github.com/fankiat/sopht-mpi/pull/141/commits/72cf577bf8b80e2e8bfcb32872c9193350eeb215)